calabash/Classes/FranklyServer/Routes/LPSuspendAppRoute.m

```diff
@@ -0,0 +1,97 @@
+#if ! __has_feature(objc_arc)
+#warning This file must be compiled with ARC. Use -fobjc-arc flag (or convert project to ARC).
+#endif
+
+#import <UIKit/UIKit.h>
+#import "LPSuspendAppRoute.h"
+#import "LPCocoaLumberjack.h"
+#import "LPInfoPlist.h"
+#import "LPInvoker.h"
+
+@interface UIApplication (LP_SUSPEND_APP_CATEGORY)
+
+- (void) suspend;
+
+@end
+
+@interface LPSuspendAppRoute ()
+
+- (CGFloat) durationWithDictionary:(NSDictionary *) arguments;
+
+@end
+
+@implementation LPSuspendAppRoute
+
+- (BOOL) supportsMethod:(NSString *) method atPath:(NSString *) path {
+  return [method isEqualToString:@"GET"];
+}
+
+- (CGFloat) durationWithDictionary:(NSDictionary *) arguments {
+  NSNumber *durationNumber = [arguments objectForKey:@"duration"];
+  if (durationNumber) {
+    return [durationNumber doubleValue];
+  } else {
+    return 2.0;
+  }
+}
+
+- (NSDictionary *) JSONResponseForMethod:(NSString *) method
+                                     URI:(NSString *) path
+                                     data:(NSDictionary *) data {
+
+  CGFloat duration = [self durationWithDictionary:data];
+
+
+  UIBackgroundTaskIdentifier __block task;
+
+  // I am not sure why this is necessary.  It does not appear to be called.
+  // I don't see the log message and I tried to force an app exit by inserting
+  // an `abort()` call.
+  task = [[UIApplication sharedApplication] beginBackgroundTaskWithName:@"Resume"
```

Comment:
On iOS Simulators, you can simply:

```
[[UIApplication sharedApplication] suspend];
< sleep >
[LSApplicationWorkspace openApplicationWithBundleID:]
```

because for app are apparently _allowed_ to perform tasks in the background on Simulators.


On devices, the story is different.  When an app is sent to the background (resigns active) it cannot execute any task - unless the app declares that it wants to perform a background task.   Hence the need for this code block.

I _think_ this method registers that the app wants to perform a background task.

The block is only called if the app exceeds the _background task timeout_?  That is where it gets murky for me.

I took this background pattern from the Apple examples.